### PR TITLE
Add guide to add auth to Vercel with open source Ory

### PR DESCRIPTION
**Pull request by aeneasr**
This PR adds a guide which explains how to add authentication with login, user management, registration, ... to NextJS deployed on Vercel. We use the [Ory Kratos](https://github.com/ory/kratos) open source identity server as the user backend.

### CODE REVIEW

**Comment by aeneasr** [comment]
@lfades anything I can do to help get this merged? :)

**Comment by okbel** [comment]
@aeneasr Hey! Can you add some context to the PR? It's empty now. I'll take a look.

**Comment by aeneasr** [comment]
Of course! I updated the opening comment :)

**Comment by okbel** [comment]
Works pretty good! Could you add the same UI as the rest of the examples? https://edge-functions-bot-protection-datadome.vercel.sh/ ? LMK if you need some help there.

**Comment by aeneasr** [comment]
Cool :) Yes, it would be great if you had 1 or 2 pointers!

**Comment by goncy** [comment]
> Cool :) Yes, it would be great if you had 1 or 2 pointers!

Hey @aeneasr thanks for the PR! For using the common layout you can see:

* [_app.tsx](https://github.com/vercel/examples/blob/main/solutions/reduce-image-bandwidth-usage/pages/_app.tsx)
* [line 3 and 62 here](https://github.com/vercel/examples/blob/main/solutions/reduce-image-bandwidth-usage/pages/index.tsx)

If you have any question just let us know :)

**Comment by aeneasr** [comment]
Thank you! :) Done :)
Screenshot 2021-12-12 at 23-34-58 Authentication with Next js Ory

**Review by lfades — CHANGES_REQUESTED** [review]
We need a new folder for this example, because it's not about using edge functions.

The example is following the Next.js template and the contents of the example don't say a lot about it. @aeneasr Can you update the example to only include what's relevant to it (remove prettier, eslint, other unrequired configs, the content of the Next.js template, etc) and also update the contents on the example in a way that it explains itself by just looking at the demo, feel free to add buttons and more text like in our existing examples.

**Comment by aeneasr** [comment]
Sorry, I am a bit confused-the commenter above said to use the edge template, should I now remove it again?

The code was following other examples in this repo, such as the Clerk one. Also what do you mean with „new folder“?

**Comment by aeneasr** [comment]
Gentle ping @lfades :) I think there are a lot of benefits of combining Ory with Next.js and Vercel. We are getting quite a lot of questions around integrating Nextjs with Ory, and our recent blog post about it was on HN front page for two days.

I think our two ecosystems ([Ory](https://github.com/ory) is the biggest open source auth ecosystem with 45M docker pulls and 30k+ stars) fit well together and Nextjs users will benefit quite a lot from this example and accompanying docs!

**Comment by okbel** [comment]
Hey @aeneasr! Sorry about the confusion. 

1) To add some clarifications: It's not using `_middleware` therefore it's not using edge functions. 
What we mean by adding it to another folder is because it's misleading to add it under `edge-functions` if it's not using edge functions (Check the other examples using _middleware and go here for more information https://vercel.com/features/edge-functions). 

Don't worry about this one. After merge I can add it to the right folder. 👍 

2) If you check this example: https://edge-functions-bot-protection-botd.vercel.sh/ or this one https://edge-functions-ip-blocking-datadome.vercel.sh/ it shows an interactive landing where it explains what the examples does and creates a flow between buttons. If you think this is done already and showcases everything for Ory then I'll move forward and merge it in a couple of hours if I don't hear back.

Thanks for your contribution! Big :fan: of Ory!

**Review by okbel — APPROVED** [review]

**Comment by aeneasr** [comment]
Got it, thank you! :) I'll try to address the feedback today :)

**Comment by aeneasr** [comment]
> To add some clarifications: It's not using _middleware therefore it's not using edge functions.
What we mean by adding it to another folder is because it's misleading to add it under edge-functions if it's not using edge functions (Check the other examples using _middleware and go here for more information https://vercel.com/features/edge-functions).

Got it :) I saw the new example pages and I think "auth" would be the best suited category :)

I've updated the pages according to your feedback! I tried to mimic the design from the others. Let me know if this works!

Screenshot 2021-12-21 at 20 09 21

Screenshot 2021-12-21 at 20 09 23

**Comment by aeneasr** [comment]
Happy new year, and a gentle ping @okbel :)

**Comment by okbel** [comment]
Happy new year! This looks amazing can @goncy or @lfades take a look? (LGTM)

File: edge-functions/auth-with-ory/pages/_app.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const Layout = getLayout<LayoutProps>(Component)
+  console.log('layout', Layout)
+
+  return (
+    <Layout
+      title="Add login and registration to your Next.js app!"
+      path="Auth, Login, Registration with Ory"
+      deployButton={{
+        repositoryUrl:
+          'https://github.com/vercel/examples/tree/main/solutions/auth-with-ory'
```
Comment (goncy) [inline edge-functions/auth-with-ory/pages/_app.tsx:16]: I think the correct url is `/edge-functions/auth-with-ory` 🤔

File: edge-functions/auth-with-ory/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { getLayout } from '@vercel/edge-functions-ui'
+import '@vercel/edge-functions-ui/globals.css'
+import type { LayoutProps } from '@vercel/edge-functions-ui/layout'
+import type { AppProps } from 'next/app'
+
+function App({ Component, pageProps }: AppProps) {
+  const Layout = getLayout<LayoutProps>(Component)
+  console.log('layout', Layout)
```
Comment (goncy) [inline edge-functions/auth-with-ory/pages/_app.tsx:8]: Oops

**Review by goncy — CHANGES_REQUESTED** [review]
This PR is awesome! Thanks 🙌, just left a few comments

File: edge-functions/auth-with-ory/pages/_app.tsx (hunk 2 above)
Comment (aeneasr) [reply edge-functions/auth-with-ory/pages/_app.tsx:8]: 👍 

```suggestion
```

File: edge-functions/auth-with-ory/pages/_app.tsx (hunk 1 above)
Comment (aeneasr) [reply edge-functions/auth-with-ory/pages/_app.tsx:16]:
```suggestion
          '/edge-functions/auth-with-ory'
```

**Review by goncy — APPROVED** [review]

**Comment by aeneasr** [comment]
Wohooo thank you! :)

**Pull request merged**